scala2-functional-program-design/week5/calculator/src/main/scala/calculator/Polynomial.scala

```diff
@@ -3,11 +3,15 @@ package calculator
 object Polynomial extends PolynomialInterface {
   def computeDelta(a: Signal[Double], b: Signal[Double],
       c: Signal[Double]): Signal[Double] = {
-    ???
+    Signal[Double](Math.pow(b.apply(),2) - 4 * a.apply() * c.apply())
   }
 
   def computeSolutions(a: Signal[Double], b: Signal[Double],
       c: Signal[Double], delta: Signal[Double]): Signal[Set[Double]] = {
-    ???
+    Signal({
+      val delta = computeDelta(a, b, c)()
+      if (delta < 0) Set()
+      else Set((-b.apply() + Math.sqrt(delta)) / (2 * a.apply()), (-b.apply() - Math.sqrt(delta)) / (2 * a.apply()))
```

Comment:
```suggestion
      else Set((-b() + Math.sqrt(delta)) / (2 * a()), (-b() - Math.sqrt(delta)) / (2 * a()))
```

Reply:
I would prefer splitting into variables, and giving proper names, seems more readable to me. Your call though